src/Common/tests/System/PlatformDetection.cs

```diff
@@ -34,6 +34,58 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static int s_isWinRT = -1;
+
+        public static bool IsWinRT
+        {
+            get
+            {
+                if (s_isWinRT != -1)
+                    return s_isWinRT == 1;
+
+                if (!IsWindows || IsWindows7)
+                {
+                    s_isWinRT = 0;
+                    return false;
+                }
+
+                byte[] buffer = new byte[0];
+                uint bufferSize = 0;
+                try
+                {
+                    int result = GetCurrentApplicationUserModelId(ref bufferSize, buffer);
+                    switch (result)
+                    {
+                        case 15703: // APPMODEL_ERROR_NO_APPLICATION
+                            s_isWinRT = 0;
+                            break;
+                        case 0:     // ERROR_SUCCESS
+                        case 122:   // ERROR_INSUFFICIENT_BUFFER
+                                    // Success is actually insufficent buffer as we're really only looking for
+                                    // not NO_APPLICATION and we're not actually giving a buffer here. The
+                                    // API will always return NO_APPLICATION if we're not running under a
+                                    // WinRT process, no matter what size the buffer is.
+                            s_isWinRT = 1;
+                            break;
+                        default:
+                            throw new InvalidOperationException($"Failed to get AppId, result was {result}.");
+                    }
+                }
+                catch (Exception e)
+                {
+                    // We could catch this here, being friendly with older portable surface area should we
+                    // desire to use this method elsewhere.
+                    if (e.GetType().FullName.Equals("System.EntryPointNotFoundException", StringComparison.Ordinal))
+                        // API doesn't exist, likely pre Win8
+                        s_isWinRT = 0;
+                    else
+                        throw;
```

Comment:
nit, this should have braces, I believe we only allow no-brace if it's a single line if.